Template files with dashes `-` produce invalid code for `moduleManifest.ts`

**Describe the bug**
Filenames with dashes make the templater produce invalid code for `moduleManifest.ts`

```
✘ [ERROR] Expected "from" but found "-"

    .mailing/src/moduleManifest.ts:9:12:
      9 │ import reset-password from "../../emails/reset-password";
        │             ^
        ╵             from

wait  - compiling...
error - unhandledRejection: Error: Build failed with 1 error:
.mailing/src/moduleManifest.ts:9:12: ERROR: Expected "from" but found "-"
    at failureErrorWithLog (/home/mokhtar/app/remixtape/node_modules/mailing/node_modules/esbuild/lib/main.js:1575:15)
    at /home/mokhtar/app/remixtape/node_modules/mailing/node_modules/esbuild/lib/main.js:1033:28
    at /home/mokhtar/app/remixtape/node_modules/mailing/node_modules/esbuild/lib/main.js:978:67
    at buildResponseToResult (/home/mokhtar/app/remixtape/node_modules/mailing/node_modules/esbuild/lib/main.js:1031:7)
    at /home/mokhtar/app/remixtape/node_modules/mailing/node_modules/esbuild/lib/main.js:1143:14
    at responseCallbacks.<computed> (/home/mokhtar/app/remixtape/node_modules/mailing/node_modules/esbuild/lib/main.js:680:9)
    at handleIncomingPacket (/home/mokhtar/app/remixtape/node_modules/mailing/node_modules/esbuild/lib/main.js:735:9)
    at Socket.readFromStdout (/home/mokhtar/app/remixtape/node_modules/mailing/node_modules/esbuild/lib/main.js:656:7)
    at Socket.emit (node:events:513:28)
    at addChunk (node:internal/streams/readable:324:12)
    at readableAddChunk (node:internal/streams/readable:297:9)
    at Readable.push (node:internal/streams/readable:234:10)
    at Pipe.onStreamRead (node:internal/stream_base_commons:190:23) {
  errors: [
    {
      detail: undefined,
      id: '',
      location: [Object],
      notes: [],
      pluginName: '',
      text: 'Expected "from" but found "-"'
    }
  ],
  warnings: []
}
```

**To Reproduce**
Steps to reproduce the behavior:

1. Create an email with a dash in the filename i.e. `reset-password.tsx`
2. Run `npx mailing preview`
3. See error in the logs

**Expected behavior**
I would expect my email templates to get picked up and work even if their file names contain dashes

**Desktop (please complete the following information):**

- OS: Manjaro Linux
- Version 0.9.10

--- a/.changeset/orange-pears-explain.md
+++ b/.changeset/orange-pears-explain.md
@@ -1,0 +1,5 @@
+---
+"mailing": patch
+---
+
+Support email templates with dashes in filename

--- a/packages/cli/src/commands/preview/server/setup.ts
+++ b/packages/cli/src/commands/preview/server/setup.ts
@@ -52,13 +52,14 @@ export async function linkEmailsDirectory(emailsDir: string) {
   const relativePathToEmailsDir = posix.relative(dotMailingSrcPath, emailsDir);
 
   uniquePreviewCollections.forEach((p) => {
-    const moduleName = p.replace(/\.[jt]sx/g, "");
+    const fileName = p.replace(/\.[jt]sx/g, "");
+    const previewModuleName = `${camelCase(fileName)}Preview`;
     previewImports.push(
-      `import * as ${moduleName}Preview from "${
-        relativePathToEmailsDir + "/previews/" + moduleName
+      `import * as ${previewModuleName} from "${
+        relativePathToEmailsDir + "/previews/" + fileName
       }";`
     );
-    previewConsts.push(`${moduleName}: ${moduleName}Preview`);
+    previewConsts.push(`"${fileName}": ${previewModuleName}`);
   });
 
   let indexFound = false;
@@ -78,11 +79,12 @@ export async function linkEmailsDirectory(emailsDir: string) {
   const templateImports: string[] = [];
   const templateModuleNames: string[] = [];
   uniqueTemplates.forEach((p) => {
-    const moduleName = p.replace(/\.[jt]sx/g, "");
-    templateModuleNames.push(moduleName);
+    const fileName = p.replace(/\.[jt]sx/g, "");
+    const moduleName = camelCase(fileName);
+    templateModuleNames.push(`"${fileName}": ${moduleName}`);
     templateImports.push(
       `import ${moduleName} from "${
-        relativePathToEmailsDir + "/" + moduleName
+        relativePathToEmailsDir + "/" + fileName
       }";`
     );
   });
@@ -251,4 +253,8 @@ async function writeFrontendManifest(toPath: string) {
   ).toString();
   await writeFile(toPath, feManifestContents);
   await buildManifest("browser", toPath);
+}
+
+function camelCase(string: string) {
+	return string.replace(/-./g, match => match[1].toUpperCase());
 }

--- a/packages/cli/src/commands/preview/server/setup.ts
+++ b/packages/cli/src/commands/preview/server/setup.ts
@@ -16,7 +16,7 @@ import {
 
 import { debug, log } from "../../../util/serverLogger";
 import { build, BuildOptions } from "esbuild";
-import { template } from "lodash";
+import { camelCase, template } from "lodash";
 import { getNodeModulesDirsFrom } from "../../util/getNodeModulesDirsFrom";
 import { lintEmailsDirectory } from "../../util/lintEmailsDirectory";
 
@@ -253,8 +253,4 @@ async function writeFrontendManifest(toPath: string) {
   ).toString();
   await writeFile(toPath, feManifestContents);
   await buildManifest("browser", toPath);
-}
-
-function camelCase(string: string) {
-	return string.replace(/-./g, match => match[1].toUpperCase());
 }

